starknet-signers/src/key_pair.rs

```diff
@@ -0,0 +1,136 @@
+use starknet_core::{
+    crypto::{ecdsa_sign, EcdsaSignError, Signature},
+    types::{H256, U256},
+};
+use starknet_crypto::{get_public_key, FieldElement};
+
+#[derive(Debug)]
+pub struct KeyPair {
+    private_key_u256: U256,
+    private_key_fe: FieldElement,
```

Comment:
I don't like this struct too much, because they are both representation so the same and one is derived from the other. Since it's a key *pair*, what about having both `private_key` and `public_key` as pub members of the struct? 
```rust
pub struct KeyPair {
    pub private_key: U256,
    pub public_key: U256
}
```

Granted, one can still be derived from the other, but at least they're different "category".

Or alternatively have a `Secret` instead of a `KeyPair` which would wrap only the private key?

Reply:
Hmm I think having the public key computed by default is kind of weird... What if the user never calls `get_public_key`? That computing would be wasted right?

Reply:
I also don't like having 2 fields of literally the same thing here... But it's opaque to users as they're private. It's there for performance reasons just in case it's not obvious.

Reply:
Maybe we change the name to `PrivateKey` or `SigningKey`? The equivalent in `ethers-rs` seems to be named `SigningKey`, and the public key type is called `VerifyingKey`.

Reply:
Can also remove the duplicated fields to make it more idiomatic, though it's not visible to users.

Reply:
Yeah, it's wasted in this sense. Do you think it's too much?

BTW it's true for `private_key_fe` as well - it's only used in `get_public_key`, though converting `U256` to `FieldElement` isn't as expensive.

Reply:
Well that isn't totally wasted as the process of converting to `FieldElement` in `new()` is also responsible for the range check.

Reply:
I like that solution of `SigningKey` and `VerifyingKey` like in `ethers-rs`.

Reply:
Cool. Lemme refactor it a bit.

Reply:
There you go. Refactored.